### ISSUE TITLE
Additional index.md disappear with navigation.indexes

### ISSUE THREAD

**Issue by fottsia** [squidfunk/mkdocs-material#3281]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Consider the following navigation in mkdocs.yml:

```
nav:
  - Section:
    - section/index.md
    - dir1/index.md
    - section/dir2/index.md
```

In this configuration, only the first index.md is displayed, while all the others are hidden.

### Expected behaviour

All the three entries should appear.

### Actual behaviour

Only the first index.md appears.

### Steps to reproduce

Create a nav: like the one described above.

### Package versions

- Python: `3.9.9`
- MkDocs: `1.2.3`
- Material: both `8.0.1` and `7.3.6`

### Configuration

```yaml
The nav: shown above.
```

### System information

- Operating system: Ubuntu 22.04
- Browser: Firefox

**Comment by alkisg** [comment]
I think the semantics of navigation.indexes would be a lot simpler if it meant this:

> When the navigation.indexes feature is enabled, the first page listed under a section is shown when that section is clicked.

Which would mean that in this case:

```
nav:
  - Section:
    - Page 1: section/page-1.md
    ...
    - Page n: section/page-n.md
    - section/index.md
    - Section/index.md
```

...when Section would be clicked, page-1.md would show up, NOT section/index.md, and NOT Section/index.md.
If someone did want Section/index.md to be the "section index page", they should list it first.

**Comment by squidfunk** [comment]
Working as intended – you can only have one `index.md` per folder. This is a limitation of MkDocs. The [section-index-plugin](https://github.com/oprypin/mkdocs-section-index) might handle those cases, though. It's not specific to Material for MkDocs, though.

**Comment by alkisg** [comment]
But it's not one folder. The index.md are in different folders.

Do you mean that a section can only have content from a single folder, not different ones?

**Comment by squidfunk** [comment]
> When the navigation.indexes feature is enabled, the first page listed under a section is shown when that section is clicked.

This is not how this works from a template perspective. Every page that is rendered from a file called `index.md` gets an ?`is_index` property that evaluates to `true`. If there ar multiple, we need to decide what to render as index, which is why we just take the first one.

https://github.com/squidfunk/mkdocs-material/blob/ae6bdb427801ac77618c22f2e00eef5e16bdb8ec/src/partials/nav-item.html#L66-L74

> But it's not one folder. The index.md are in different folders.

Yes, but we're talking about the navigation, and when you list them on the same layer, it appears to the theme that it is in the same folder. Nothing we can do about it.

**Comment by squidfunk** [comment]
Workaround: just rename the files so that you have on `index.md` per navigation level.

**Comment by alkisg** [comment]
> ...we need to decide what to render as index, which is why we just take the first one.

OK, but why do you need to remove the others? I don't know about theme engines, can't the theme store a boolean variable "this page was used as an auto-index", and INCLUDE the pages that don't have that var == true?

> Workaround: just rename the files so that you have on index.md per navigation level.

But that would make it impossible to have clickable section1 > section2, right? Because only one of them would be allowed to have an index.md...

**Comment by squidfunk** [comment]
The issue was fixed in f3926bf44.

**Comment by squidfunk** [comment]
Released as part of 8.1.3.
